src/Xrefcheck/Verify.hs

```diff
@@ -259,11 +259,17 @@ checkExternalResource VerifyConfig{..} link
             Right () -> return $ Right ()
             Left   _ -> makeRequest GET 0.7
   where
-    isIgnored = any (\l ->
-        let (b, m, a) = link =~ l :: (Text, Text, Text)
-        in null b && null a && not (null m)) vcIgnoreRefs
+    isIgnored =
+        let mIsIgnored = (doesMatchAnyRegex link) <$> vcIgnoreRefs
+        in maybe True id mIsIgnored
+
     doesReferLocalhost = any (`T.isInfixOf` link) ["://localhost", "://127.0.0.1"]
 
+    doesMatchAnyRegex :: Text -> ([Text] -> Bool)
+    doesMatchAnyRegex txt = any $ \t ->
+        let (before, match, after) = txt =~ t :: (Text, Text, Text)
```

Comment:
Some more comments:

1. I heard from Java world that compiling regex pattern may be an expensive operation, it is better to do once and then apply multiple times.
2. Regex compilation can fail.

Taking these two into account, maybe let's store not `Text`s in `vsIngoreRefs` rather [`Regex`](https://hackage.haskell.org/package/regex-tdfa-1.3.1.0/docs/Text-Regex-TDFA-Text.html#t:Regex)s, and implement a JSON instance that compiles the pattern, reporting an appropriate error if any occurs? Failing fast sounds like a good thing.

The `Config.hs` module even has an example of defining `FromJSON` instance manually, the implementation for `Regex`  should look pretty similar.